### DISCUSSION TITLE
Python extension not inferring implicit function call

### DISCUSSION THREAD

**Discussion by AHJiapengZhang** [microsoft/pylance-release#6713] [Q&A]
<img width="1064" alt="image" src="https://github.com/user-attachments/assets/08bd5203-aeb2-4cef-8dd3-94eb6a18f8d1">

In this example, the Python extension is not identifying implicit calling from Python partial function. I suspect this doesn't work for decoraters as well.

**Comment by loveinabox** [comment]
Maybe it is related to a feature within the Python programming language, specifically concerning the `functools.partial` function and the code analysis capabilities of an Integrated Development Environment (IDE). The `functools.partial` is used to create a new function that, when called, passes the given arguments to the original function.
In the provided code snippet, a function `function(a, b, c)` is defined, and then `partial` is used to create a new function `f`, which has the arguments `a=1` and `b=2` pre-set. When `f(3)` is called, it is essentially calling `function(1, 2, 3)`.
If the Python extension ( likely referring to the code analysis tools within the IDE) does not recognize this implicit call, it may be because it has not correctly analyzed the behavior of the `partial` function. This could potentially affect features such as code completion and error detection.

**Reply by AHJiapengZhang** [reply]
You elaborated well! I think this also applies to other functions that take a function as a parameter and function decorators as well(though I haven't tried yet).
